cart/application/cartService_test.go

```diff
@@ -399,6 +400,183 @@ func TestCartService_AdjustItemsToRestrictedQty(t *testing.T) {
+				session: web.EmptySession().Store(cartApplication.GuestCartSessionKey, "some_guest_id"),
+				force:   false,
+			},
+			want: "201910251128792ZM",
+		},
+		{
+			name: "force reserved order id, reserved before",
+			fields: fields{
+				CartReceiverService: func() *cartApplication.CartReceiverService {
+					result := &cartApplication.CartReceiverService{}
+					result.Inject(
+						new(MockGuestCartServiceWithModifyBehaviour),
+						new(MockCustomerCartService),
+						func() *decorator.DecoratedCartFactory {
+							result := &decorator.DecoratedCartFactory{}
+							result.Inject(
+								&MockProductService{},
+								flamingo.NullLogger{},
+							)
+
+							return result
+						}(),
+						&authApplication.AuthManager{},
+						&authApplication.UserService{},
+						flamingo.NullLogger{},
+						nil,
+						&struct {
+							CartCache cartApplication.CartCache `inject:",optional"`
+						}{
+							CartCache: new(MockCartWithItemCache),
+						},
+					)
+					return result
+				}(),
+				ProductService: &MockProductService{},
+				Logger:         flamingo.NullLogger{},
+				EventPublisher: new(MockEventPublisher),
+				config: &struct {
+					DefaultDeliveryCode string `inject:"config:commerce.cart.defaultDeliveryCode,optional"`
+					DeleteEmptyDelivery bool   `inject:"config:commerce.cart.deleteEmptyDelivery,optional"`
+				}{
+					DefaultDeliveryCode: "default_delivery_code",
+					DeleteEmptyDelivery: false,
+				},
+				DeliveryInfoBuilder: new(MockDeliveryInfoBuilder),
+				CartCache:           new(MockCartWithItemCache),
+				RestrictionService:  nil,
+				PlaceOrderService:   &MockPlaceOrderService{},
+			},
+			args: args{
+				ctx:     context.Background(),
+				session: web.EmptySession().Store(cartApplication.GuestCartSessionKey, "some_guest_id"),
+				force:   true,
+			},
+			want: "201910251128788TD",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cs := &cartApplication.CartService{}
+			authmanager := &authApplication.AuthManager{}
+			authmanager.Inject(
+				flamingo.NullLogger{},
+				nil, nil, nil,
+			)
+			cs.Inject(
+				tt.fields.CartReceiverService,
+				tt.fields.ProductService,
+				tt.fields.EventPublisher,
+				tt.fields.EventRouter,
+				tt.fields.DeliveryInfoBuilder,
+				tt.fields.RestrictionService,
+				authmanager,
+				tt.fields.Logger,
+				tt.fields.config,
+				&struct {
+					CartValidator     validation.Validator      `inject:",optional"`
+					ItemValidator     validation.ItemValidator  `inject:",optional"`
+					CartCache         cartApplication.CartCache `inject:",optional"`
+					PlaceOrderService placeorder.Service        `inject:",optional"`
+				}{
+					PlaceOrderService: tt.fields.PlaceOrderService,
+				},
+			)
+
+			var got *cartDomain.Cart
+			if tt.args.force == true {
+				got, _ = cs.ForceReserveOrderIDAndSave(tt.args.ctx, tt.args.session)
+			} else {
+				got, _ = cs.ReserveOrderIDAndSave(tt.args.ctx, tt.args.session)
```

Comment:
It would be nice if there would be a test for each of these functions. Additional cases for when there is no OrderID reserved yet would also be great.